pkg/types/builder.go

```diff
@@ -293,6 +293,14 @@ func (g *Builder) generateTypeName(suffix string, names ...string) (string, erro
 	if g.Package.Scope().Lookup(n) == nil {
 		return n, nil
 	}
+	// start from 2 considering the 1st of this type is the one without an
+	// index.
+	for i := 2; i < 10; i++ {
```

Comment:
I wonder if we can get away with appending the CRD kind name repeatedly. Instead of `SubnetworkParameters2`, how would `SubnetworkParameters_Subnetwork` or `Subnetwork_SubnetworkParameters` look? Feels like because they are in the same package, it's kind of too easy to confuse the one used by one resource with the other when it's only a number that differentiates them.

Reply:
Given this is just generated code and we don't expect anyone to deal with modifying those types, I prefer numbers.
One case could be people importing these types to interact programmatically but still, in that case, I believe using the wrong type would cause compilation errors hence easy to figure out.

Even if we go with appending the kind name as you suggested, we will still have both `SubnetworkParameters` and `SubnetworkParameters_Subnetwork` in the same package and the former would belong to `RouterNat` resource instead of `Subnetwork` which is still confusing.

Reply:
I was thinking about someone importing this package and using the types rather than modifying. In such cases, the types being generated or not doesn't really matter for them and Go types having numbers in the name doesn't look very idiomatic. I'm OK with both approaches though.